org.eclipse.gef/src/org/eclipse/gef/dnd/TemplateTransferDropTargetListener.java

```diff
@@ -73,13 +73,15 @@ protected final CreateRequest getCreateRequest() {
 	 * @param template the template Object
 	 * @return a Factory
 	 */
+	@SuppressWarnings("static-method") // allow children to override this method
 	protected CreationFactory getFactory(Object template) {
-		if (template instanceof CreationFactory) {
-			return ((CreationFactory) template);
-		} else if (template instanceof Class) {
-			return new SimpleFactory((Class) template);
-		} else
-			return null;
+		if (template instanceof CreationFactory creationFactory) {
+			return creationFactory;
+		}
+		if (template instanceof Class clazz) {
+			return new SimpleFactory(clazz);
```

Comment:
Similar to the case in ShapeEditor, I'd use something like this to resolve the warning:
```
if (template instanceof Class<?> clazz) {
	return new SimpleFactory<>(clazz);
}
```

Reply:
Done